packages/toggle-model/src/InCollectionMatchingSegment.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pheature\Model\Toggle;
+
+use Pheature\Core\Toggle\Read\Segment;
+
+class InCollectionMatchingSegment implements Segment
+{
+    public const NAME = 'in_collection_matching_segment';
+    private string $id;
+    /** @var array<string, mixed> */
+    private array $criteria;
+
+    /**
+     * InCollectionMatchingSegment constructor.
+     *
+     * @param string $id
+     * @param array<string, mixed> $criteria
+     */
+    public function __construct(string $id, array $criteria)
+    {
+        $this->id = $id;
+        $this->criteria = $criteria;
+    }
+
+    public function id(): string
+    {
+        return $this->id;
+    }
+
+    public function criteria(): array
+    {
+        return $this->criteria;
+    }
+
+    public function match(array $payload): bool
+    {
+        /** @var mixed $criterionValue */
+        foreach ($this->criteria as $field => $criterionValue) {
+            if (!array_key_exists($field, $payload)) {
+                return false;
+            }
+
+            if (!$this->isAMatch($payload[$field], $criterionValue)) {
+                return false;
+            }
+        }
+
+        return true;
```

Comment:
When `$this->criteria` is not present, we will return true.

I don't know if it is the expected behavior. Maybe if we haven't any criteria it should be false, because we can't match the given payload against anything.

Reply:
You're totally right, it makes more sense returning `false` when the `$this->criteria` is not present. I'm going to fix that now!

Reply:
done and nice catch!!